src/osd/pg_scrubber.h

```diff
@@ -413,6 +413,10 @@ class PgScrubber : public ScrubPgIF, public ScrubMachineListener {
 
   [[nodiscard]] bool is_scrub_active() const final { return m_active; }
 
+  [[nodiscard]] bool get_reserve_failed() const final { return m_reserve_failed; }
```

Comment:
Should be part of the m_local_osd_resource object.

Seems to me that the clearing API shouldn't exist.

Reply:
This flag represents a temporary state of the PG.  When other scrubs finish, reservations can succeed. This flag is a way to cycle through all possible scrubs to try to find one that can run now.  We have to keep trying in order, so after 1 cycle we clear all flags to start over.